projects/refer-and-monitor-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/NsiService.kt

```diff
@@ -30,14 +35,47 @@ class NsiService(
     private val nsiOutcomeRepository: NsiOutcomeRepository,
     private val statusHistoryRepository: NsiStatusHistoryRepository,
     private val contactRepository: ContactRepository,
-    private val contactTypeRepository: ContactTypeRepository
+    private val contactTypeRepository: ContactTypeRepository,
+    private val createNsi: CreateNsi
 ) : AuditableService(auditedInteractionService) {
+
+    @Transactional
+    fun startNsi(crn: String, rs: ReferralStarted) = audit(MANAGE_NSI) { audit ->
+        val nsi = nsiRepository.findByPersonCrnAndExternalReference(crn, rs.urn)
+            ?: createNsi.new(crn, rs).apply {
+                statusHistoryRepository.save(statusHistory())
+                contactRepository.save(contact(NSI_REFERRAL.value, referralDate))
+                contactRepository.save(statusChangeContact())
+                contactRepository.save(contact(NSI_COMMENCED.value, actualStartDate!!))
+                audit["offenderId"] = person.id
+                audit["nsiId"] = id
```

Comment:
Is it worth adding these to the audit context even if we're updated an existing NSI?